src/psyclone/domain/lfric/lfric_extract_driver_creator.py

```diff
@@ -492,8 +492,18 @@ def _create_output_var_code(name, program, is_input, read_var,
                 # If it is not indexed then `name` will already end in "_data"
                 post_tag = f"{name}{postfix}"
         name_lit = Literal(post_tag, CHARACTER_TYPE)
-        BaseDriverCreator.add_call(program, read_var,
-                                   [name_lit, Reference(post_sym)])
+
+        if sym.is_array and not sym.datatype.is_allocatable:
+            # In case of a non-allocatable array (e.g. a constant
+            # size array from a module), call the ReadVariable
+            # function that does not require an allocatable field
+            BaseDriverCreator.add_call(program, read_var+"NonAlloc",
```

Comment:
I should know this but confess I can't remember: for arrays that are declared in modules within LFRic, does the driver just populate their values?

Reply:
Yes, their values will be stored in the kernel data file and so the arrays will be initialised.